src/main/java/bio/terra/service/filedata/azure/tables/TableDirectoryDao.java

```diff
@@ -219,11 +224,11 @@ public FireStoreDirectoryEntry retrieveByPath(
       TableServiceClient tableServiceClient,
       UUID collectionId,
       String tableName,
-      List<String> fullPaths) {
+      Set<String> fullPaths) {
     List<TableEntity> entities =
         fullPaths.stream()
             .map(FileMetadataUtils::makeLookupPath)
-            .distinct()
+            .distinct() // Still need this b/c could pass in path both with and without prefix
```

Comment:
It's just my opinion, but I prefer not using end-of-line comments, as they make the code hard to edit and reformat. In most cases the comment works the same if put on the line before, instead of at the end of the line.